internal/services/managedhsm/internal.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package managedhsm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type deleteAndPurgeNestedItem interface {
+	DeleteNestedItem(ctx context.Context) (autorest.Response, error)
+	NestedItemHasBeenDeleted(ctx context.Context) (autorest.Response, error)
+	PurgeNestedItem(ctx context.Context) (autorest.Response, error)
+	NestedItemHasBeenPurged(ctx context.Context) (autorest.Response, error)
```

Comment:
we should be removing all references to autorest, so instead can we return at the very least an http response instead?